### ISSUE TITLE
Connection reset causes app to exit

### ISSUE THREAD

**Issue by mattgodbolt** [mattgodbolt/compiler-explorer#228]
Not sure what causes it, but saw this in the log a few minutes before a Pingdom alert:
```
Jan 08 08:32:53 5eae40a39aaf gcc:  events.js:160 
Jan 08 08:32:53 5eae40a39aaf gcc:        throw er; // Unhandled 'error' event 
Jan 08 08:32:53 5eae40a39aaf gcc:        ^ 
Jan 08 08:32:53 5eae40a39aaf gcc:  Error: read ECONNRESET 
Jan 08 08:32:53 5eae40a39aaf gcc:      at exports._errnoException (util.js:1022:11) 
Jan 08 08:32:53 5eae40a39aaf gcc:      at Pipe.onread (net.js:569:26) 
Jan 08 08:32:53 5eae40a39aaf gcc:  Sun Jan 08 2017 14:32:53 GMT+0000 (UTC) 
Jan 08 08:32:53 5eae40a39aaf gcc:  Program node app.js --env amazon --port 10240 --lang C++ --static out/dist --archivedVersions /opt/compiler-explorer-archive exited with code 1 
```

**Comment by mattgodbolt** [comment]
Has been happening a lot it seems; can find a ton of instances in the log.

**Comment by mattgodbolt** [comment]
Still seems to be happening:

```
Jan 08 22:23:11 c9a01a62a582 gcc:  events.js:160 
Jan 08 22:23:11 c9a01a62a582 gcc:        throw er; // Unhandled 'error' event 
Jan 08 22:23:11 c9a01a62a582 gcc:        ^ 
Jan 08 22:23:11 c9a01a62a582 gcc:  Error: read ECONNRESET 
Jan 08 22:23:11 c9a01a62a582 gcc:      at exports._errnoException (util.js:1022:11) 
Jan 08 22:23:11 c9a01a62a582 gcc:      at Pipe.onread (net.js:569:26) 
Jan 08 22:23:11 c9a01a62a582 gcc:  Mon Jan 09 2017 04:23:12 GMT+0000 (UTC) 
Jan 08 22:23:11 c9a01a62a582 gcc:  Program node app.js --env amazon --port 10240 --lang C++ --static out/dist --archivedVersions /opt/compiler-explorer-archive exited with code 1 
```

Though the text `Error: ...` shouldn't appear in the code any more (should now be `Caught error:` or `Promise error:` ... unless I'm confused about date/times (which I don't think I am given I just got the email alert and it's about the right time). Must mean the error is coming from elsewhere...

**Comment by mattgodbolt** [comment]
Stream errors seem to be addressed by 0d04f0b437a26c40342f98cd057192a988bb5b29, but the ECONNRESET stuff's still going on.

**Comment by mattgodbolt** [comment]
Finally fixed! Seems if the far child dies in some cases it appears as a connreset. Currently logging as error, maybe should be warning only.
